### DISCUSSION TITLE
Do you use `git pull --rebase`?

### DISCUSSION THREAD

**Discussion by iBug** [squidfunk/mkdocs-material#6903] [General]
I see there are a lot of [Merge branch 'master' of github.com:squidfunk/mkdocs-material](https://github.com/squidfunk/mkdocs-material/commits/master/) commits, and many of these are merges of a single divergent commit. IMO these single-commit merges are most of the time only clutters to the commit history, and I personally have developed a habit of using `git pull --rebase` (short option `-r`, config `pull.rebase = true`) so the Git history stays linear. I wonder if you find this tip useful. @squidfunk

**Comment by squidfunk** [comment]
Thanks for asking. I was always not entirely sure what the benefits of merge vs rebase is, but as I understand, rebase will stop at every commit if there are conflicts, right? Merging them locally, I always thought it was easier to fix them in one go, but I've wanted to give this a go anyway. It just got lost among everything else 😅
